ui/app/components/app/modals/qr-scanner/qr-scanner.component.js

```diff
@@ -107,37 +113,59 @@ export default class QrScanner extends Component {
 
 
     // Ethereum address links - fox ex. ethereum:0x.....1111
-    if (content.split('ethereum:').length > 1) {
-
-      type = 'address'
-      values = {'address': content.split('ethereum:')[1] }
+    if (content.split('eths:/?s=').length > 1) {
+    // signature from external signer
+      type = 'signature'
+      values = {'signature': content.split('eths:/?s=')[1]}
 
     // Regular ethereum addresses - fox ex. 0x.....1111
+    } else if (content.match(/^\s*[a-fA-F0-9]{130}\s*^/)) {
+      type = 'signature'
+      values = {'signature': content}
     } else if (content.substring(0, 2).toLowerCase() === '0x') {
-
       type = 'address'
       values = {'address': content }
-
+    } else if (content.match(/^\s*ethereum:\?signature=[a-fA-F0-9]{130}\s*^/)) {
```

Comment:
If we want to use the QR code scanner tool as a generic thing, the schema-specific parts should happen somewhere else

Reply:
Yep I will rewrite it.

Reply:
Done. https://github.com/MetaMask/metamask-extension/pull/6267/commits/550c4c662f3a67dc55dc39df8931fac3562af269